### PR TITLE
Install DB and apply patches in a single layer

### PR DESCRIPTION
**Pull request by thelateperseus**
The oracle-docker-images repository allows patches to be applied, but the resulting image size is doubled (see https://github.com/oracle/docker-images/issues/1990). Instead, we would like to apply the patches in the same layer where the database binaries are installed.

I have applied this only to the 19c and 21c Dockerfiles since the other versions are either at or rapidly approaching end of life.

Changes:
- Set SLIMMING to false to ensure that all files required for patching are present.
- Copied patchDBBinaries.sh, runDatapatch.sh and savePatchSummary.sh from the extensions/patch directory.
- Modified patchDBBinaries.sh to delete temporary files to reduce image size.
- Added steps to builder stage in Dockerfile to copy the patches to the container and run the patchDBBinaries.sh script.
- Added steps to Dockerfile to copy datapatch and lspatches scripts under $ORACLE_BASE.

### CODE REVIEW

**Comment by gvenzl** [comment]
Hi @thelateperseus, thanks a lot for your contribution! Please sign the Oracle Contributor Agreement outlined above for us to be able to consider this PR.

**Comment by thelateperseus** [comment]
Hi @gvenzl, I signed the OCA yesterday, but it's still "Under review".

**Comment by thelateperseus** [comment]
Hi @gvenzl, Looks like my OCA signature has been verified now.

**Comment by gvenzl** [comment]
Great, thanks a lot, @thelateperseus!

@yunus-qureshi, can you please review this PR, thanks!

**Comment by yunus-qureshi** [comment]
@thelateperseus - We are evaluating this PR. Can you please address the lint issues.

File: OracleDatabase/SingleInstance/dockerfiles/19.3.0/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ LABEL "provider"="Oracle"                                               \
       "port.oemexpress"="5500"
 
 # Argument to control removal of components not needed after db software installation
-ARG SLIMMING=true
+ARG SLIMMING=false
```
[the remark this answers is not in the document]
Comment (thelateperseus) [reply OracleDatabase/SingleInstance/dockerfiles/19.3.0/Dockerfile]: I've removed the SLIMMING arg and also the extensions/patches directory.

**Comment by thelateperseus** [comment]
@yunus-qureshi, I believe I've fixed everything requested. Please let me know if I missed anything.

**Comment by aditya059** [comment]
Hi @thelateperseus 

Thanks for your contribution but we are going with a different approach. I have raised a PR for the same. 

There are some disadvantages of your PR:
1. We have to build the container image from scratch every time we want to apply a release update or one-offs. By scratch, I mean we have to place the linux base image (LINUX.X64_193000_db_home.zip) and release update and/or one-offs in patch folder and run the build script (buildContainerImage.sh). This will increase the build time. 
2. We will not be able to extend (or patch) any previously build image (or patch image). E.g. Images in container registry cannot be patched with your PR.

**Comment by thelateperseus** [comment]
Thanks @aditya059. I don't mind the approach taken as long as the resulting image size is reduced.

I assume you're referring to #2682. I tried using your branch to build a 19.3 image then patch it to 19.16. The 19.3 image was 7.9GB and the 19.16 image was 7.77GB, so it looks like it's working as expected.

**Comment by aditya059** [comment]
Closing this PR as the other PR (#2682) is merged.

**Pull request closed without merging**